File: bankingOnGolang/internal/model/valueObjects.go

package model

import (
    "github.com/shopspring/decimal"
)

 type Money struct {
     Amount            decimal.Decimal
    Currency            string

// parent associations as their child

}

 type Address struct {
     Street            string
    City            string
    State            string
    PostalCode            string
    Country            string

// parent associations as their child

}

 Name is value
type AccountNumber     string

 Name is value
type IBAN     string

 Name is value
type BIC     string

 Name is value
type CardPAN     string

 Name is value
type Percentage     decimal.Decimal

File: bankingonaspdotnet/Domain/ValueObjects.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace bankingonaspdotnet.Domain;

[ComplexType]
public record Money(
decimal Amount,
string Currency
);

[ComplexType]
public record Address(
string Street,
string City,
string State,
string PostalCode,
string Country
);

[ComplexType]
public record AccountNumber(
string Value
);

[ComplexType]
public record IBAN(
string Value
);

[ComplexType]
public record BIC(
string Value
);

[ComplexType]
public record CardPAN(
string Value
);

[ComplexType]
public record Percentage(
decimal Value
);

